tool/lib/test/unit.rb

```diff
@@ -915,10 +925,20 @@ def setup_options(opts, options)
           options[:most_asserted] = n
         end
         opts.on '--launchable-test-reports=PATH', String, 'Report test results in Launchable JSON format' do |path|
-          require 'json'
-          options[:launchable_test_reports] = writer = JsonStreamWriter.new(path)
-          writer.write_array('testCases')
-          at_exit{ writer.close }
+          if Test::Unit::AutoRunner::Runner === self
```

Comment:
If this option makes sense only for the specific class, this option should belong to that class.

Reply:
I think the stuffs for launchable can be a new module, and `Test::Unit::AutoRunner::Runner` should include it.

Reply:
Are you referring to something like this: https://github.com/ruby/ruby/pull/9777/commits/001a3a8003f974df792e1df481dd43b35ba4d56c#diff-d58358cd56895151702a7ae7aae9bc0891f2e692a147a46f388e3abb784e354bR1905? I believe achieving this might be challenging because Test::Unit::Worker invokes the setup_options method, but the configuration for `--launchable-test-reports` is not set. This discrepancy arises because the `setup_options` method in `Test::Unit::LaunchableOption` responsible for configuring --launchable-test-reports is not invoked, resulting in a failure with an "invalid option" message like https://github.com/ruby/ruby/actions/runs/7779280421/job/21210179713#step:12:145.

Reply:
Thus, I changed like this [7fc4679](https://github.com/ruby/ruby/pull/9777/commits/7fc467910378a6a60d20fe5884c8d1c77f880e59)